tests/test_slim_engine.py

```diff
@@ -1420,17 +1422,118 @@ def test_no_mutation_types_defined(self):
+            id="test",
+            mutation_types=[mt, mt],
+            proportions=[0.5, 0.5],
+            description="test test",
+            long_description="test test test",
+        )
+        contig.add_dfe(
+            intervals=np.empty((0, 2), dtype="int"),
+            DFE=dfe,
+        )
+        engine = stdpopsim.get_engine("slim")
+        with pytest.raises(ValueError):
+            engine.simulate(
+                demographic_model=self.model,
+                contig=contig,
+                samples=self.samples,
+                extended_events=extended_events,
+                dry_run=True,
+            )
+
+    def test_duplicate_mutation_ids(self):
+        extended_events = [
+            stdpopsim.ext.DrawMutation(
+                time=self.T_mut,
+                mutation_id="test",
+                population_id=0,
+                coordinate=100,
+            ),
+        ]
+        contig = get_test_contig()
+        contig.add_single_site_mutation_type(id="test")
+        contig.add_single_site_mutation_type(id="test")
+        engine = stdpopsim.get_engine("slim")
+        with pytest.raises(ValueError):
+            engine.simulate(
+                demographic_model=self.model,
+                contig=contig,
+                samples=self.samples,
+                extended_events=extended_events,
+                dry_run=True,
+            )
+
+    def test_mutation_has_interval(self):
+        extended_events = [
+            stdpopsim.ext.DrawMutation(
+                time=self.T_mut,
+                mutation_id="test",
+                population_id=0,
+                coordinate=100,
+            ),
+        ]
+        contig = get_test_contig()
+        contig.add_single_site_mutation_type(id="test")
+        contig.interval_list[1] = np.array([[99, 101]], dtype="int")
+        engine = stdpopsim.get_engine("slim")
+        with pytest.raises(ValueError):
+            engine.simulate(
+                demographic_model=self.model,
+                contig=contig,
+                samples=self.samples,
+                extended_events=extended_events,
+                dry_run=True,
+            )
+
+    def test_mutation_with_oor_mutation_type_id(self):
+        draw_mut = stdpopsim.ext.DrawMutation(
+            time=self.T_mut,
+            mutation_id="test",
+            population_id=0,
+            coordinate=100,
+        )
+        delattr(draw_mut, "mutation_id")
+        setattr(draw_mut, "mutation_type_id", -1)
+        contig = get_test_contig()
+        engine = stdpopsim.get_engine("slim")
+        with pytest.raises(ValueError):
+            engine.simulate(
+                demographic_model=self.model,
+                contig=contig,
+                samples=self.samples,
+                extended_events=[draw_mut],
+                dry_run=True,
+            )
+
     def test_bad_time(self):
         for time in (-1,):
             with pytest.raises(ValueError):
```

Comment:
here we really need a `match=XXX` because this could *also* throw the error that there's no such mutation_id.